test/unit/org/apache/cassandra/db/commitlog/CommitLogUpgradeTest.java

```diff
@@ -102,6 +103,27 @@ public void cleanUp()
         Assert.assertEquals("JVM killed", shouldBeKilled, killerForTests.wasKilled());
     }
 
+    // 30 matches version in MessagingService, 3.0.13 is the latest patch release after 3.0.0 but before 3.0.14
+    @Test
+    public void test30_encrypted() throws Exception
+    {
+        testRestore(DATA_DIR + "3.0.13-encrypted");
+    }
+
+    // 3014 matches version in MessagingService, 3.0.29 is the latest patch release after 3.0.14
+    @Test
+    public void test3014_encrypted() throws Exception
+    {
+        testRestore(DATA_DIR + "3.0.29-encrypted");
+    }
+
+    // 40 matches version in MessagingService, 4.0.11 is the latest patch release on 4.0
+    @Test
+    public void test40_encrypted() throws Exception
+    {
+        testRestore(DATA_DIR + "4.0.11-encrypted");
+    }
+
```

Comment:
nit: how can we fail when new MS versions are added but there's no corresponding test…?